test/test.cpp

```diff
@@ -24,3 +24,49 @@
 // For more information, please refer to <http://unlicense.org/>
 
 #include "utf8.h"
+
+#if __cplusplus >= 202002L
+using char_type = char8_t;
+#else
+using char_type = char;
+#endif
+
+
+// We don't care about the results. We only want to check compilation
+
+constexpr void test()
+{
+    constexpr char_type in_str[20]{};
```

Comment:
See my comment below - that should let you replace `char_type` with `utf8_int8_t` instead here I hope?

Reply:
Yes, it would work